ReactWindows/ReactNative.Shared/Tracing/LogTracerBase.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Diagnostics.Tracing;
+using System.Text;
+
+namespace ReactNative.Tracing
+{
+    /// <summary>
+    /// Generic interface to provide log tracer functionality
+    /// Used by <see cref="RnLog"/> class to trace log messages.
+    /// The interface is designed to be able to integrate other logging
+    /// frameworks like nlog, log4net, etc
+    /// </summary>
+    public abstract class LogTracerBase
```

Comment:
Should this also be disposable to better ensure flush is called?

Reply:
I could add a Dispose that does Flush, and I guess it's the app;s responsibility to call it (which means this only applies to WPF apps).
I don't think I can have Dispose be called by Finalizer, it's too late.

---
In reply to: [221401652](https://github.com/Microsoft/react-native-windows/pull/1987#discussion_r221401652) [](ancestors = 221401652)